### PR TITLE
feat: retrieve immutable content via nrs

### PR DESCRIPTION
**Pull request by jacderida**
- 2b06d802c **refactor: nrs map fetch and rename multimap**

  The `MultimapKeyValues` type alias was renamed to `Multimap` to make it easier to work with. Some
  functions were also renamed to the same effect, e.g., `fetch_multimap_values` -> `fetch_multimap`.
  
  The function for retrieving the NRS map was also refactored. Some aspects were broken down into
  smaller functions to reduce some complex if/else cases. Things were also setup for the version in
  the last entry in the Multimap to be collected. This is going to be used with the returned
  NrsMapContainer, to facilitate returning links to immutable content, which doesn't have a version.
  Currently the code isn't setup to support that scenario.

- e87774f48 **feat: retrieve immutable content via nrs**

  Changes the resolver to enable retrieval of NRS links to immutable content. Previously, the
  NRS `target_url` was incorrectly being checked to see if it contained a version. We now validate
  this URL using the same process used by NRS, which won't assert that a file link has a version. To
  reduce duplication, the `validate_nrs_url` function in the NRS module was changed to public.

  Tests were added to both the API and CLI to cover the scenario.

  The NrsMap struct was extended to include a subname_version field. This is going to be used to
  request the map with a particular version of a subname. If no version was specified when the
  container was retrieved, then the field won't be set. This is why it's an Option. Since we're
  storing this field on the NrsMap, the `version` field on the `NrsMapContainer` was removed.

  A couple of CLI NRS tests were also re-enabled, one of which happened to be related to immutable
  content.

### CODE REVIEW

**Review by bochaco — APPROVED** [review]

File: sn_api/src/app/nrs/nrs_map.rs [hunk 1]
```diff
@@ -22,9 +23,13 @@ pub(crate) type SubName = String;
 /// | "sub"         | "sub.example"    | "safe://eg2" |
 /// | "sub.sub"     | "sub.sub.example"| "safe://eg3" |
 ///
+/// The map also contains a version, which is the latest version in the Multimap where this map is
```
Comment (grumbach) [inline sn_api/src/app/nrs/nrs_map.rs:26]: What is this version refering to? Subnames are versionned but not the map itself right?

**Review by grumbach — APPROVED** [review]
LGTM! (assuming the version in the NRS map is going to be moved inside the map entry keys in a follow up PR)

**Pull request merged**